### DISCUSSION TITLE
Fetching multiple API queries in nextjs getStaticProps() method

### DISCUSSION THREAD

**Discussion by sagharfrancis** [vercel/next.js#22518] [Help]
I have a next js project with apollo and the graphql.

I am fetching data from a CMS.

I have separate graphql queries to fetch the main navigation and the footer items.

The queries are in the src/queries folder. I am using these queries in index.js.

`import client from '../src/apollo/ApolloClient';
import navbar from '../src/quries/navbar.js';
import footer from '../src/quries/footer.js';
import Layout from '../src/components/layout/Layout';
import React from 'react'

export default function Home({ data }) {
  return (
    <React.Fragment>
      <Layout data={data}>

      </Layout>
    </React.Fragment>
  );
}

export const getStaticProps = async () => {
  const { data } = await client.query({
    query: navbar,
  });
  return {
    props: {
          headermenu: data.menu.edges
    },
    revalidate: 1,
  };
};`

the problem is I am able to fetch only one query at a time like above I fetched navbar. My target is to fetch both the navbar and footer in the getStaticProps() method. If there any other alternative rather than passing both to getStaticProps()

**Comment by smontlouis** [comment]
You can fetch both queries and returning them:

```js
export const getStaticProps = async () => {
  const { data: navbarData } = await client.query({
    query: navbar,
  });

  const { data: footerData } = await client.query({
      query: footer,
  });

  return {
    props: {
      headermenu: navbarData.menu.edges,
      footer: navBarFooter.foo.bar
    },
    revalidate: 1,
  };
};`
```


or using `Promise.all`

```js
export const getStaticProps = async () => {
  const navbarQuery = client.query({
    query: navbar,
  });

  const footerQery = client.query({
      query: footer,
  });

  const responses = await Promise.all([navQuery, footerQuery])

  return {
    props: {
      headermenu: responses[0].data.menu.edges,
      footer: responses[1].data.foo.bar
    },
    revalidate: 1,
  };
};`
```

**Reply by sagharfrancis** [reply]
Thank you soo much for helping. My issue is resolved.

**Reply by fz3hra** [reply]
is there a way to write the queries in a single query please?

**Comment by ghost** [comment]
very good. thank's a lot... : )

**Comment by Shreyas-29** [comment]
Can we use getStaticProps() multiple times in index.js ? ie. in single file?
